Editor/src/Windows/InspectorWindow.bf

```diff
@@ -0,0 +1,316 @@
+
+			if (!inspector._fileBuffer.IsEmpty)
+				inspector._fileBuffer.Clear();
+
+			if (inspector._filePath == null)
+				inspector._filePath = new .(filePath);
+			else
+				inspector._filePath.Set(filePath);
+
+			File.ReadAllText(filePath, inspector._fileBuffer);
+		}
+
+		private void ClearFileView()
+		{
+			_isFileView = false;
+			_fileBuffer.Clear();
+		}
+
+		public override void OnRender()
+		{
+			if (_isFileView)
+			{
+				ShowFileView();
+				return;
+			}
+
+			if (_entity == null)
+				return;
+
+			_entity.IsEnabled = EditorGUI.Checkbox("##EntityEnabled", _entity.IsEnabled);
+
+			_entityName.Clear();
+			Editor.GetEntityName(_entity.Id, _entityName);
+
+			EditorGUI.SameLine();
+			EditorGUI.ItemWidth(-25);
+			if (EditorGUI.Input("##EntityName", _entityName).OnChange)
+				Editor.SetEntityName(_entity.Id, _entityName);
+
+			EditorGUI.AlignFromRight(25);
+			if (EditorGUI.Button("X"))
+			{
+				Application.Instance.[Friend]RemoveEntity(_entity);
+				SetCurrentEntity(null);
+				Editor.InvalidateSave();
+				return;
+			}	
+
+			EditorGUI.Line();
+			EditorGUI.AddColumns();
+
+			var componentsToRender = scope List<BaseComponent>();
+
+			for (var component in Editor.Instance.[Friend]_components.Values)
+				if (component.Parent == _entity)
+					componentsToRender.Add(component);
+
+			for (var component in componentsToRender)
+			{
+				var componentName = scope String();
+				component.GetType().GetShortName(componentName);
+				if (componentName.EndsWith("Component"))
+				{
+					componentName.RemoveFromEnd(9);
+					componentName.[Friend]Realloc(componentName.AllocSize);
+				}
+
+				EditorGUI.ItemID(scope String()..AppendF("{}", component));
+				var isOpen = EditorGUI.BeginCollapsableHeader(componentName);
+
+				EditorGUI.AlignFromRight(20);
+				if (EditorGUI.Selectable("X"))
+				{
+					_entity.RemoveComponent(component);
+					continue;
+				}
+
+				if (isOpen)
+				{
+					RenderObject(component);
+					EditorGUI.EndCollapsableHeader();
+				}
+			}
+
+			EditorGUI.RemoveColumns();
+			EditorGUI.NewLine();
+			EditorGUI.AlignMiddle(130);
+			if (EditorGUI.Button("Add Component", .(130, 20)))
+				ImGui.OpenPopup("Add Component");
+
+			ShowAddComponentPopup();
+		}
+
+		private void ShowFileView()
+		{
+			EditorGUI.ItemWidth(EditorGUI.GetWindowWidth() - 23);
+			EditorGUI.Input("##FilePath", _filePath, "", 256, true);
+
+			EditorGUI.SameLine();
+			if (EditorGUI.Button("R"))
+				ViewFile(_filePath);
+
+			EditorGUI.DisableItem();
+			EditorGUI.InputMultiline("##FileView", _fileBuffer, 1024, true, EditorGUI.GetWindowSize());
+		}
+
+		private void ShowAddComponentPopup()
+		{
+			Type componentType = null;
+
+			if (ImGui.BeginPopup("Add Component"))
+			{
+				for (var type in Type.Types)
+				{
+					if (type.IsSubtypeOf(typeof(BaseComponent)))
+					{
+						var typeName = scope String();
+						type.GetShortName(typeName);
+						if (typeName == "BaseComponent" || typeName == "BehaviourComponent")
```

Comment:
The type name is `BehaviorComponent` (US spelling), not `BehaviourComponent` (UK spelling).

Is there a way to use reflection to get a `nameof` like in C#? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/nameof

Reply:
Btw, because of this misspelling, an error shows up in the console when trying to add a `BehaviorComponent`